src/core/public/chrome/chrome_service.tsx

```diff
@@ -98,13 +98,35 @@ export class ChromeService {
 
     const appTitle$ = new BehaviorSubject<string>('Kibana');
     const brand$ = new BehaviorSubject<ChromeBrand>({});
-    const isVisible$ = new BehaviorSubject(true);
     const isCollapsed$ = new BehaviorSubject(!!localStorage.getItem(IS_COLLAPSED_KEY));
     const applicationClasses$ = new BehaviorSubject<Set<string>>(new Set());
     const helpExtension$ = new BehaviorSubject<ChromeHelpExtension | undefined>(undefined);
     const breadcrumbs$ = new BehaviorSubject<ChromeBreadcrumb[]>([]);
     const badge$ = new BehaviorSubject<ChromeBadge | undefined>(undefined);
 
+    /**
+     * These observables allow consumers to toggle the chrome visibility via either:
+     *   1. Using setIsVisible() to trigger the next visibility$
+     *   2. Setting `chromeHidden` when registering an application, which will
+     *      reset the visibility whenever the next application is mounted
+     *   3. Having embed=true in the query string
+     */
+    const visibility$ = new BehaviorSubject(true);
+    const isVisible$ = merge(
```

Comment:
Using `merge` here makes it so that the last observable to emit is going to be the one that used in the `map` on line 126. This seems like a subtle race condition. Maybe we should define the behavior so that if `chromeHidden` is true, the manually set visibility is ignored. In other words, `chromeHidden` takes precedence over `visibility$`.

This would prevent edge cases, but it would also prevent some other plugin from toggling the Chrome UI to visible when the login page is displaying (which would not make any sense).

This would have the caveat that a app with `chromeHidden: true` would not be able to toggle the Chrome UI to visible, but I also don't think that's a use-case we need to support.

Reply:
I refactored the way the observables work here, instead of tracking *visibility*, the logic seems simpler to determine if the chrome is *hidden*. I think this also means this "race" doesn't happen, if anything is telling the chrome to be hidden, nothing can override it to be shown.